clicommand/agent_start.go

```diff
@@ -214,13 +214,13 @@ var AgentStartCommand = cli.Command{
 		},
 		cli.BoolFlag{
 			Name:   "disconnect-after-job",
-			Usage:  "Disconnect the agent after running a job",
+			Usage:  "Disconnect the agent as soon as the job they're running completes",
```

Comment:
What do you think about "Disconnect the agent after running exactly one job"?

(Aside: I think the agent has been “it” so far, probably never “they”. I guess that could be open to change?)

Reply:
The big question in my mind around these flags is: how does it interact with `--spawn=N` when the one agent process has multiple agent worker connections?

I'm not sure whether or not this is the right place to try and clarify that.

Reply:
so after digging into this a bit, the interaction is that these flags work on an agent-worker basis, rather than an agent basis - that is, this flag actually means "disconnect the agent after all of the workers it runs have completed exactly one job". 

this is kinda confusing, butttttttt i think it's okay to leave the interactions between `--spawn` and the flags whose docs i'm changing here, because it's unlikely(?) that they'll be used in concert. the upshot is that `--spawn 5 --disconnect-after-job` could be used as a "run exactly this many jobs and then exit, i suppose 🤷

Reply:
the other thing we could do is just encode this further context in the help text - 
```
when used in conjunction with the `--spawn` flag, each worker booted will run exactly one job.
```
... or something